spec/takeaway_spec.rb

```diff
@@ -0,0 +1,51 @@
+require 'takeaway'
+
+describe Takeaway do
+
+  subject { Takeaway.new(sms_sender) }
+  let(:sms_sender) { double(:sms_sender) }
+
+  it "responds to #menu"do
+    expect(subject).to respond_to(:menu)
+  end
+
+  describe "#read_menu" do
+
+    it "should have contents" do
+      expect(subject.menu.empty?).to be false
+    end
+
+  end
+
+  it "should respond to #place_order" do
+    expect(subject).to respond_to(:place_order).with(3).arguments
+  end
+
+  describe "#place order" do
+
+    before(:each) do
+      @order_list = { "Grilled banana" => 2 }
+    end
+
+    it "should raise error if item not on menu" do
+      bad_order_list = { "Koala Pie" => 4 }
+      expect { subject.place_order(bad_order_list, "£20", :phonenumber) }.to raise_error "Koala Pie not on menu."
+    end
+
+    it "should raise_error if expected order total is incorrect" do
+      expect { subject.place_order(@order_list, 6, :phonenumber) }.to raise_error "Expected total incorrect."
```

Comment:
Metrics/LineLength: Line is too long. [109/80] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)